statsd/statsd.go

```diff
@@ -332,11 +333,15 @@ func newWithWriter(w statsdWriter, o *Options, writerName string) (*Client, erro
 		metrics:   &ClientMetrics{},
 	}
 	// Inject values of DD_* environment variables as global tags.
+	extraTags := []string{}
 	for envName, tagName := range ddEnvTagsMapping {
 		if value := os.Getenv(envName); value != "" {
-			c.Tags = append(c.Tags, fmt.Sprintf("%s:%s", tagName, value))
+			extraTags = append(extraTags, fmt.Sprintf("%s:%s", tagName, value))
 		}
 	}
+	// We sort env tags to ease testing since map order access is not guarantee
+	sort.Strings(extraTags)
```

Comment:
Is this necessary? The tests seem to sort tags anyway. I'm not sure how perf-critical this code is, but if we do need tags in specific order, perhaps we could convert `ddEnvTagsMapping` to be a slice instead. Seems unnecessary to keep sorting what is basically a constant.